examples/adiabatic/linear.py

```diff
@@ -38,13 +39,14 @@ def main(nqubits, hfield, T, dt, solver, save):
     """
     h0 = hamiltonians.X(nqubits)
     h1 = hamiltonians.TFIM(nqubits, h=hfield)
+    bac = h1.backend
 
     # Calculate target values (H1 ground state)
     target_state = h1.ground_state()
-    target_energy = h1.eigenvalues()[0].numpy().real
+    target_energy = bac.to_numpy(h1.eigenvalues()[0]).real
 
     # Check ground state
-    state_energy = callbacks.Energy(h1)(target_state).numpy()
+    state_energy = bac.to_numpy(callbacks.Energy(h1).apply(bac, target_state)).real
```

Comment:
> for instance, the `callback.apply` looks a bit cumbersome to me

I agree, it is not very good. I believe 
```suggestion
    state_energy = bac.to_numpy(h1.expectation(target_state)).real
```
would work in this case, right?

Reply:
How can I do the same thing for the `Overlap` callback?

Reply:
There is no, so I think for the overlap only the current solution with `apply` works. Alternatively, you could also just calculate the overlap directly:
```py
np.dot(np.conj(state1), state2)
```
but if the current solution works there is no issue.